projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid.pipes.ts

```diff
@@ -39,7 +40,8 @@ export class IgxTreeGridHierarchizingPipe implements PipeTransform {
         }
 
         grid.flatData = grid.transactions.enabled ?
-        flatData.filter(rec => !grid.transactions.getState(this.getRowID(primaryKey, rec))) : flatData;
+        flatData.filter(rec => !grid.transactions.getState(this.getRowID(primaryKey, rec)) ||
+            grid.transactions.getState(this.getRowID(primaryKey, rec)).type !== TransactionType.ADD) : flatData;
```

Comment:
I would suggest to cache the `grid.transactions.getState(this.getRowID(primaryKey, rec))` value as a `const` in the arrow function's body.

Reply:
Done